### ISSUE TITLE
Bug on deploy: Error response from daemon: Cannot kill container 

### ISSUE THREAD

**Issue by nielsonsantana** [dokku/dokku#1961]
Some times I getting this problem and can't do deploy. 
It's happen after I deploy my project for several times.

Erro log:
remote: + docker inspect 906b849bd033ec6b510fe724afb44366926ecec2d6427665eecaa9a8acbcdf4e
remote: + docker stop 906b849bd033ec6b510fe724afb44366926ecec2d6427665eecaa9a8acbcdf4e
remote: + docker kill 906b849bd033ec6b510fe724afb44366926ecec2d6427665eecaa9a8acbcdf4e
remote: Error response from daemon: Cannot kill container 906b849bd033ec6b510fe724afb44366926ecec2d6427665eecaa9a8acbcdf4e: notrunning: Container 906b849bd033ec6b510fe724afb44366926ecec2d6427665eecaa9a8acbcdf4e is not running
remote: Error: failed to kill containers: [906b849bd033ec6b510fe724afb44366926ecec2d6427665eecaa9a8acbcdf4e]

To /home/dokku/website-production/
 ! [remote rejected] master -> master (pre-receive hook declined)
Process exited with: 1. Reason was:  ()

Output of the following commands
- `uname -a`:
    Linux dokku-production 3.13.0-57-generic #95-Ubuntu SMP Fri Jun 19 09:28:15 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
r
- `free -m`:

   ```
total             used        free      shared    buffers     cached
Mem:          2001       1860        141          5        144        892
-/+ buffers/cache:        823       1178
Swap:         1023         66        957
```

- `docker version`:
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:12:04 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:12:04 UTC 2015
 OS/Arch:      linux/amd64

- `docker -D info`:
Containers: 16
Images: 70
Server Version: 1.9.1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 102
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.13.0-57-generic
Operating System: Ubuntu 14.04.3 LTS
CPUs: 2
Total Memory: 1.955 GiB
Name: dokku-production
ID: ZCTL:26E7:G77H:YQHS:6LEB:34QI:7EPU:VO64:652T:36V5:Y4DT:D6PQ
WARNING: No swap limit support

- `sigil -v`:
- `docker run -ti gliderlabs/herokuish:latest herokuish version`:
- `dokku version`:
v0.4.11-4-gc79a43f
- `dokku plugin`:
deploy-hooks         0.2.0 enabled    support for pre- and post-deploy hooks
apt                  0.4.0 enabled    Inject deb packages into dokku based on files in project
  
- `docker inspect CONTAINER_ID` (if applicable):
  (BEWARE: `docker inspect` will print environment variables for some commands, be sure you're not exposing any sensitive information when posting issues. You may replace these values with XXXXXXX):

- `cat /home/dokku/<app>/nginx.conf` (if applicable):
- Link to the exact repository being deployed (if possible/applicable):
- If a deploy is failing:
My application is in Python with Django. I use the dokku deploy way
- Output of failing dokku commands after running `dokku trace on`

Environment details (AWS, VirtualBox, physical, etc.):
DigitalOcean 2GB
How was dokku installed?:
The instalation of dokku was through script.

How reproducible:
This problem happened after I do a lot of deploys. 
The deploy was did in normal flow(two times a week) but during 4 months,
until the problem happens.
It's occur in 3 projects on my server.

**Comment by josegonzalez** [comment]
What do you mean by "do a lot of deploys"? Do you mean lots of consecutive deploys? Deploying quickly within a specific time-frame? Or just if you do it in general?

**Comment by michaelshobbs** [comment]
What was above those lines though? Those errors are coming from the zero downtime feature attempting to stop/kill the failed new container.

**Comment by nielsonsantana** [comment]
Sorry, the deploy was did in normal flow(two times a week) but during 4 months, 
until the problem happens.
It's occur with 3 projects on my server.

**Comment by josegonzalez** [comment]
Can you gist the entire log output after enabling `dokku trace on`? Just be sure to clear out any secrets in that gist.

**Comment by nielsonsantana** [comment]
Clear my secrets datas in the log need a lot of work . :/
The link for the log: https://gist.github.com/nielsonsantana/667505ecda1a91ab9e32

**Comment by josegonzalez** [comment]
Did you change anything recently with the system? It seems like the container checks are saying that your containers failed to start...

**Comment by nielsonsantana** [comment]
The changes that I did was add two plugins, but only `apt`:
`deploy-hooks` 0.2.0 enabled support for pre- and post-deploy hooks
`apt` 0.4.0 enabled Inject deb packages into dokku based on files in project

Hummm.. The problem is that the docker can't kill the old container e can't start the new.
My application is running fine. I have other enviromnent in staging. Also, I created other project with same source code and worked fine.

**Comment by michaelshobbs** [comment]
@nielsonsantana these lines are actually the root of your issue.
https://gist.github.com/nielsonsantana/667505ecda1a91ab9e32#file-erro-log-tracert-on-dokku-txt-L8118-L8131

The new container dokku tried to spin up with your app did not stay up. The error to which you are referring is superfluous output of dokku's attempt to clean up after itself.

**Comment by nielsonsantana** [comment]
hummm... Great point. But what should I do? Increase the time to wait the application start?
I checked the log of my application and the web service is running.

    [2016-03-03 00:52:59 +0000] [11] [INFO] Starting gunicorn 19.2.1
    [2016-03-03 00:52:59 +0000] [11] [INFO] Listening at: http://0.0.0.0:5000 (11)
    [2016-03-03 00:52:59 +0000] [11] [INFO] Using worker: eventlet
    [2016-03-03 00:52:59 +0000] [184] [INFO] Booting worker with pid: 184
    [2016-03-03 00:52:59 +0000] [185] [INFO] Booting worker with pid: 185
    [2016-03-03 00:52:59 +0000] [186] [INFO] Booting worker with pid: 186
    [2016-03-03 00:52:59 +0000] [187] [INFO] Booting worker with pid: 187
    [2016-03-03 00:53:00 +0000] [188] [INFO] Booting worker with pid: 188

**Comment by michaelshobbs** [comment]
I would probably use a `CHECKS` file.
http://dokku.viewdocs.io/dokku/checks-examples/#checks-examples

You had included some output in your original report but cut off the part that should the output of the newly started container. Can you share the entire deploy log without trace enabled?

**Comment by nielsonsantana** [comment]
Follow the log with `trace off`
    https://gist.github.com/nielsonsantana/d835744d31ddb77089ed

**Comment by nielsonsantana** [comment]
I changed the time from 10 to 30 seconds to wait the application start.

**Comment by michaelshobbs** [comment]
This is your error
```
remote: [1G       setuidgid: usage: setuidgid account child        
```

What in your `Procfile`?

**Comment by nielsonsantana** [comment]
In my procfile I have it:

    web: gunicorn --pythonpath mamatest wsgi:application --worker-class eventlet --workers 5

**Comment by michaelshobbs** [comment]
Yeah your web proc starts up fine but what about theses?

```
remote: [1G=====> worker_sync_s3=1        
remote: [1G=====> run_migrations=1        
```

**Comment by josegonzalez** [comment]
@nielsonsantana If you changed the timing for application start, then you will also need to update the dokku checks wait timeout as documented here: http://dokku.viewdocs.io/dokku/checks-examples/

**Comment by nielsonsantana** [comment]
I will check the check-examples, thanks.
The Procfile of the application that I send for you has only the one worker:

    web: gunicorn --pythonpath mamatest wsgi:application --worker-class eventlet --workers 5 

This procfile that you have ben posted is from other project:

	web: gunicorn wsgi:application
	worker_sync_s3: s4cmd sync --sync-check static s3://$AWS_BUCKET_NAME & python worker_sync_static_s3.py
	run_migrations: python manage.py migrate

**Comment by michaelshobbs** [comment]
I'm going off the output you sent in that gist....

**Comment by michaelshobbs** [comment]
So all entries in the `Procfile` are expected to run in the foreground by default. It is unclear as to what specific problem we're troubleshooting now. Can you clarify?

**Comment by nielsonsantana** [comment]
Ok. Trying clarify the problem. 
My application `website` has a Procfile with one `web`(https://gist.github.com/nielsonsantana/d835744d31ddb77089ed#file-erro-log-tracert-off-dokku-txt-L217) and has no DOKKU_SCALE file in the repository.  
However,(I don't know how), but DOKKU_SCALE of other project was copied to `website` and there three entries presents:

    web=1
    worker_sync_s3=1
    run_migrations=1    

I deleted the file named DOKKU_SCALE on `/home/dokku/website/` and run the deploy again.
The output that I have is that the problem persist. There nothing related with Procfile and DOKKU_SCALE. 
Link for the new logs:
https://gist.github.com/nielsonsantana/4d42ea8c546011be2e3e#file-erro-log-tracert-off-dokku-and-no-dokku-scale-txt-L225-L227  
 
The problem that I have related persist.

remote: Error: No such image or container: de0d3f1d1b0c274a66139f18d1c87622984a27a810634e2765e1b59737183249

**Comment by josegonzalez** [comment]
What is the output for:

```
dokku logs website-production
```

**Comment by nielsonsantana** [comment]
The output is it:
    [2016-03-03 00:52:59 +0000] [11] [INFO] Starting gunicorn 19.2.1
    [2016-03-03 00:52:59 +0000] [11] [INFO] Listening at: http://0.0.0.0:5000 (11)
    [2016-03-03 00:52:59 +0000] [11] [INFO] Using worker: eventlet
    [2016-03-03 00:52:59 +0000] [184] [INFO] Booting worker with pid: 184
    [2016-03-03 00:52:59 +0000] [185] [INFO] Booting worker with pid: 185
    [2016-03-03 00:52:59 +0000] [186] [INFO] Booting worker with pid: 186
    [2016-03-03 00:52:59 +0000] [187] [INFO] Booting worker with pid: 187
    [2016-03-03 00:53:00 +0000] [188] [INFO] Booting worker with pid: 188

**Comment by nielsonsantana** [comment]
I tried stop the application to retry the deploy.
A get this output:
Error: No such image or container: de0d3f1d1b0c274a66139f18d1c87622984a27a810634e2765e1b59737183249
Stopping website-production ...
-----> Setting config vars
       DOKKU_APP_RESTORE: 0

**Comment by josegonzalez** [comment]
Okay, can you try a deploy, then gist the output of:

```
docker ps -a
```

**Comment by nielsonsantana** [comment]
Here de log of `docker ps -a`
Seems that I have much zombie workers.
https://gist.github.com/nielsonsantana/dbeb01791adcc4e8619c

**Comment by josegonzalez** [comment]
What were the deploy logs in this case? It might help speed up debugging if you hop onto our irc channel.

**Comment by nielsonsantana** [comment]
Can you tell me how can I sign on dokku irc channel? Link?

**Comment by josegonzalez** [comment]
http://dokku.viewdocs.io/dokku/getting-started/where-to-get-help/

**Comment by michaelshobbs** [comment]
Seems to have been solved on IRC.

**Comment by oelmekki** [comment]
I reached this page looking to solve an issue I had with deploy failing 
while still booting containers properly. Nginx configuration was not 
updated correctly to link to container, and previous containers for the 
same app were not stopped.

I don't know if it's the exact same issue, but if it isn't, I hope you 
won't mind me posting in this issue in case of other people reach that 
page too.

My problem was that I added a service in the Procfile, then (I think) 
stopped the service manually with `docker stop` or screwing my 
containers any other way, then removed the service from the Procfile.

What happened was that there was a 
`/home/dokku/<app>/CONTAINER.<old_service>.1` file that was not cleaned 
up as it should have been. At the end of deploy, dokku tried to remove 
the container, but could not find it, causing any deploy to fail with 
the symptoms described above.

The fix was to manually remove the
`/home/dokku/<app>/CONTAINER.<old_service>.1` file.
